src/elements/Asset.php

```diff
@@ -1857,7 +1857,7 @@ public function getDimensions(): ?string
      */
     public function getPath(?string $filename = null): string
     {
-        return $this->folderPath . ($filename ?: $this->_filename);
+        return $this->_volume->getFsSubpath() . $this->folderPath . ($filename ?: $this->_filename);
```

Comment:
Asset paths should continue to be relative to the root of their volume, not the filesystem.